packages/expo-cli/src/credentials/views/AndroidCredentials.ts

```diff
@@ -0,0 +1,206 @@
+
+  shouldRefetch: boolean = true;
+
+  constructor(experience: string, credentials: AndroidCredentials | null) {
+    this.experience = experience;
+    if (credentials && credentials.experienceName) {
+      this.shouldRefetch = false;
+      this.experienceName = credentials.experienceName;
+      this.keystore = credentials.keystore;
+      this.pushCredentials = credentials.pushCredentials;
+    }
+  }
+
+  async open(ctx: Context): Promise<IView | null> {
+    if (this.shouldRefetch) {
+      const appCredentials: AndroidCredentials = await ctx.api.getAsync(`credentials/android/${this.experience}`);
+      this.experienceName = get(appCredentials, 'experienceName');
+      this.keystore = get(appCredentials, 'keystore');
+      this.pushCredentials = get(appCredentials, 'pushCredentials')
+    }
+    if (!this.experienceName) {
+      this.experienceName = `@${ctx.user.username}/${this.experience}`;
+    }
+
+    if (isEmpty(this.keystore) && isEmpty(this.pushCredentials)) {
+      log(`No credentials available for ${this.experience} experience.\n`);
+    } else if (this.experienceName) {
+      log();
+      await displayAndroidAppCredentials({
+        experienceName: this.experienceName,
+        keystore: this.keystore,
+        pushCredentials: this.pushCredentials,
+      });
+      log();
+    }
+
+    const { action } = await prompt([
+      {
+        type: 'list',
+        name: 'action',
+        message: 'What do you want to do?',
+        choices: [
+          { value: 'update-keystore', name: 'Update upload keystore' },
+          { value: 'update-fcm-key', name: 'Update FCM api key' },
+          { value: 'fetch-keystore', name: 'Download keystore from expo servers' },
+          // { value: 'fetch-public-cert', name: 'Extract public cert from keystore' },
+          // {
+          //   value: 'fetch-private-signing-key',
+          //   name:
+          //     'Extract private signing key (required when migration to App Signing by Google Play)',
+          // },
+        ],
+      },
+    ]);
+
+    return this.handleAction(ctx, action);
+  }
+
+  handleAction(context: Context, selected: string): IView | null {
+    switch (selected) {
+      case 'update-keystore':
+        this.shouldRefetch = true;
+        return new UpdateKeystore(this.experience);
+      case 'update-fcm-key':
+        this.shouldRefetch = true;
+        return new UpdateFcmKey(this.experience);
+      case 'fetch-keystore':
+        return new DownloadKeystore(this.experience, this.keystore);
+      case 'fetch-public-cert':
+        return null;
+    }
+    return null;
+  }
+}
+
+class UpdateKeystore implements IView {
+  experience: string;
+
+  constructor(experience: string) {
+    this.experience = experience;
+  }
+
+  async open(ctx: Context): Promise<IView | null> {
+    const keystore = await this.provideOrGenerate(ctx);
+    await ctx.api.putAsync(`credentials/android/keystore/${this.experience}`, { keystore });
```

Comment:
Don't we need to url-encode `this.experience`?

Reply:
https://docs.expo.io/versions/latest/workflow/configuration/#slug

slug can't contain anything that needs to be escaped